### PR TITLE
Include era symbol information in data

### PR DESCRIPTION
**Pull request by Manishearth**
First step of https://github.com/unicode-org/icu4x/issues/1312

This hardcodes an era map; this is basically the only option we have if we're going to use an era code scheme.

### CODE REVIEW

**Comment by Manishearth** [comment]
This does not handle era variants; that's something that can be decided later.

File: provider/testdata/data/cldr/out [hunk 1]
```diff
@@ -0,0 +1,5351 @@
+--- cldr-cal-buddhist-full/main/ar/ca-buddhist.json	2021-11-19 12:34:54.970618693 -0800
```
Comment (sffc) [inline provider/testdata/data/cldr/out:1]: Please delete this file

**Review by sffc — DISMISSED** [review]
LGTM but please remove the extra file

**Review by sffc — APPROVED** [review]

File: components/calendar/src/gregorian.rs [hunk 2]
```diff
@@ -121,13 +121,13 @@ impl DateTime<Gregorian> {
...
         types::Year {
-            era: types::Era(tinystr8!("ce")),
+            era: types::Era(tinystr8!("ad")),
             number: year.0,
             related_iso: year.0,
         }
     } else {
         types::Year {
-            era: types::Era(tinystr8!("bce")),
+            era: types::Era(tinystr8!("bc")),
```
Comment (nordzilla) [inline components/calendar/src/gregorian.rs:130]: > question

Why are we not using `TinyStr4` for these?

**Review by nordzilla — APPROVED** [review]
LGTM, just one question.

File: components/calendar/src/gregorian.rs (hunk 2 above)
Comment (Manishearth) [reply components/calendar/src/gregorian.rs:130]: Japanese era codes are bigger

**Pull request merged**